### ISSUE TITLE
Not able to login with account migrated from Mojang to Microsoft

### ISSUE THREAD

**Issue by smilodon79** [Sintinium/oauth#21]
Minecraft Version:  1.12.2
Account type:  Microsoft (formerly Mojang, but migrated successfully)
Java Version:  jdk1.8.0_281 (shouldn't matter technically, but here goes :))

This is a weird problem.  I'm able to use oauth to login with my younger kid's account (bought as a Microsoft account only a week or so back).

I'm able to login with my Mojang account successfully.

I'm not able to login with my older kid's account (which was originally a Mojang account, but has since been migrated to Microsoft).

1.  Logging in with MultiMC works fine.  I'm able to create a single server world WITHOUT oauth and play without any issues.
2. Logging in with OAuth causes issues.  It always reports as offline (I can get you the network trace if you want)
3. I'm available on Discord if you'd like to root cause the issue
4. I don't see any standout issues in my latest.log or debug.log

Appreciate some help with this please.

**Comment by Sintinium** [comment]
Could you try it on the latest 1.16 or 1.18 versions? 1.12.2 is quite a few versions old at this point. And make sure the account's date of birth is 18 years or older because that is a requirement for some reason. If nether of those work we can try working it out on Discord. Thanks! :)

**Comment by smilodon79** [comment]
1.  Unfortunately, my favorite modpack (mc-eternal) is on 1.12
2.  Yes, this account is a minor account, but as the parent, I’ve given
permission to use Java and Minecraft.  It’s working without any problems
through multimc and the same Microsoft account.  So I’m pretty sure Minor
account isn’t the issue.

**Comment by Sintinium** [comment]
I was just asking about newer versions because they give more detail and they could also fix the problem. If they did I would just backport them. But it's interesting that the other launchers work but OAuth doesn't.

**Comment by smilodon79** [comment]
@Sintinium , I'm able to login without issues with my little kid's account.  It was bought as a Microsoft account and it's also a minor account.

I'm only having issues with my older kid's account which was migrated from Mojang to Microsoft.

I'd really appreciate if we could have RCA this issue to a fix.

**Comment by Sintinium** [comment]
The only way I can help is if you try on 1.16/1.18. You don't need a modpack or anything just oauth. 1.12 doesn't log properly so there's no way to tell what's wrong with it. Other than that I won't have much time until Tuesday to work on it

**Comment by smilodon79** [comment]
Hi
I've managed to solve the issue.  I set up a 1.18 server and tried connecting with the client.  That showed me that the issue isn't with authentication (single player worked fine), but with authorization (multiplayer was disabled).  Googling this issue told me that I needed to do the following:

1. Enable multiplayer in XBox privacy settings
2. Delete the profile in MultiMC and add again (or at least refresh the profile in MultiMC).  Without this, the changes don't get reflected and we keep getting the issue.

Finally, I'd appreciate if you're able to differentiate this issue based on the exception message and give appropriate messages, so future users are able to better handle this problem.
